mmv1/third_party/terraform/utils/config.go.erb

```diff
@@ -265,54 +265,62 @@ var DefaultClientScopes = []string{
 	"https://www.googleapis.com/auth/userinfo.email",
 }
 
-func HandleSDKDefaults(d *schema.ResourceData) {
+func HandleSDKDefaults(d *schema.ResourceData) error {
 	if d.Get("impersonate_service_account") == "" {
-		d.Set("impersonate_service_account", MultiEnvSearch([]string{
+		d.Set("impersonate_service_account", MultiEnvDefault([]string{
 			"GOOGLE_IMPERSONATE_SERVICE_ACCOUNT",
-		}))
+		}, nil))
 	}
 
 	if d.Get("project") == "" {
-		d.Set("project", MultiEnvSearch([]string{
+		d.Set("project", MultiEnvDefault([]string{
 			"GOOGLE_PROJECT",
 			"GOOGLE_CLOUD_PROJECT",
 			"GCLOUD_PROJECT",
 			"CLOUDSDK_CORE_PROJECT",
-		}))
+		}, nil))
 	}
 
 	if d.Get("billing_project") == "" {
-		d.Set("billing_project", MultiEnvSearch([]string{
+		d.Set("billing_project", MultiEnvDefault([]string{
 			"GOOGLE_BILLING_PROJECT",
-		}))
+		}, nil))
 	}
 
 	if d.Get("region") == "" {
-		d.Set("region", MultiEnvSearch([]string{
+		d.Set("region", MultiEnvDefault([]string{
 			"GOOGLE_REGION",
 			"GCLOUD_REGION",
 			"CLOUDSDK_COMPUTE_REGION",
-		}))
+		}, nil))
 	}
 
 	if d.Get("zone") == "" {
-		d.Set("zone", MultiEnvSearch([]string{
+		d.Set("zone", MultiEnvDefault([]string{
 			"GOOGLE_ZONE",
 			"GCLOUD_ZONE",
 			"CLOUDSDK_COMPUTE_ZONE",
-		}))
+		}, nil))
 	}
 
-	if d.Get("user_project_override") == "" {
-		d.Set("user_project_override", MultiEnvSearch([]string{
+	if _, ok := d.GetOk("user_project_override"); !ok {
```

Comment:
This means that we'll defer to the preexisting value when `ok` is false, right?

Reply:
We'll defer to the preexisting value when `ok` is true, right? Yup, I've tested it locally and have seen headers when I have `USER_PROJECT_OVERRIDE` set and not seen them when I have `USER_PROJECT_OVERRIDE` unset.

I'll run a test quick where I set `user_project_override` on the provider to make sure that's still ok too.